src/client/tile.cpp

```diff
@@ -765,25 +807,36 @@ void TextureSource::rebuildImagesAndTextures()
 	for (TextureInfo &ti : m_textureinfo_cache) {
 		if (ti.name.empty())
 			continue; // Skip dummy entry
+		rebuildTexture(driver, ti);
+	}
+}
 
-		video::IImage *img = generateImage(ti.name);
+void TextureSource::rebuildTexture(video::IVideoDriver *driver, TextureInfo &ti)
+{
+	if (ti.name.empty())
+		return; // this shouldn't happen, just a precaution
+
+	// replaces the previous sourceImages
+	// shouldn't really need to be done, but can't hurt
+	std::set<std::string> sourceImageNames;
```

Comment:
1. Variable names should use `snake_case` (reference https://dev.minetest.net/Code_style_guidelines#Miscellaneous )
2. Unless this is multithreaded ; wouldn't it be easier to move the `std::set` to a private `TextureSource` member variable? That avoids passing through the variable for each call.

Reply:
Will correct 1.
2, while correct, complicates any attempt to multi-thread the code in future, and gives an incorrect idea of the nature of the variable. The variable is passed by reference (as it has to be) so no significant CPU load occurs.